mlir/lib/Dialect/MIOpen/utility/builderUtils.cpp

```diff
@@ -72,6 +85,8 @@ Value createZeroConstantOp(OpBuilder &b, Location loc, Type type) {
 
   if (elementType == b.getIntegerType(8)) {
     return createConstantI8Op(b, loc, type, elementType, 0);
+  } else if (elementType == b.getIntegerType(32)) {
```

Comment:
... hold on, we can use the same trick as with `createConstantFloatOp` to make the code generic over integer types.

Reply:
Again, nice catch! I wouldn't know there's a `APInt` exist without you poking at it. Addressed in adb57e3